### ISSUE TITLE
Flatbutton ignores global textTheme inside Button.bar and AlertDialog

### ISSUE THREAD

**Issue by lazylazyllama** [flutter/flutter#44047]
## Steps to Reproduce

1. Create a simple Flatbutton
2. Set your buttonTheme in ThemeData in your global theme to:

```dart
buttonTheme: ButtonThemeData(
    textTheme: ButtonTextTheme.accent,
),
```
3. You will see, that the textcolor is white in dark mode.
4. if you set the textTheme property to ButtonTextTheme.accent, inside your FlatButton constructure, it is working:
```dart
FlatButton(
    textTheme: ButtonTextTheme.accent,
    child: Text("Text"),
 )
```

So Flatbutton is ignoring the global textTheme property, which is not good.

**Comment by iapicca** [comment]
Hi @lazylazyllama 
does this [code ](https://gist.github.com/iapicca/10e7f52f1e1e4d69680bc5ca19243de3)represent the situation you are describing?
thank you

**Comment by lazylazyllama** [comment]
@iapicca After some playing around I actually figured, that its usually working. So your code is working as it should. But it is not working if you put a Flatbutton within a Button.bar or a AlertDialog.

**Comment by TahaTesser** [comment]
Hi @lazylazyllama 

### Code Sample
```
import 'package:flutter/material.dart';

void main() => runApp(
      MyApp(),
    );

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

ThemeData _theme1 = ThemeData(
  buttonTheme: ButtonThemeData(
    textTheme: ButtonTextTheme.normal,
  ),
);
ThemeData _theme2 = ThemeData(
  buttonTheme: ButtonThemeData(
    textTheme: ButtonTextTheme.accent,
  ),
);

class _MyAppState extends State<MyApp> {
  ThemeData _themeData = _theme1;

  void _switch() => setState(() =>
      _themeData == _theme1 ? _themeData = _theme2 : _themeData = _theme1);

  @override
  Widget build(BuildContext context) => MaterialApp(
        theme: _themeData,
        home: Scaffold(
          body: Center(
            child: ButtonBar(
              alignment: MainAxisAlignment.center,
              children: [
                FlatButton(
                  onPressed: () {},
                  child: Text("Text"),
                )
              ],
            ),
          ),
          floatingActionButton:
              FloatingActionButton(onPressed: () => _switch()),
        ),
      );
}
```

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, v1.18.0-dev.5.0, on Mac OS X 10.15.4 19E287, locale
    en-GB)
    • Flutter version 1.18.0-dev.5.0 at /Users/taha/Code/flutter_dev
    • Framework revision 7f56b53de4 (4 days ago), 2020-04-12 12:00:01 -0400
    • Engine revision beb8a7ec48
    • Dart version 2.8.0 (build 2.8.0-dev.20.0 89b0f67261)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.
[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.9.1
[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
[✓] VS Code (version 1.44.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1
[✓] Connected device (4 available)
    • SM M305F   • 32003c30dc19668f • android-arm64  • Android 10 (API 29)
    • macOS      • macOS            • darwin-x64     • Mac OS X 10.15.4 19E287
    • Chrome     • chrome           • web-javascript • Google Chrome
      81.0.4044.113
    • Web Server • web-server       • web-javascript • Flutter Tools
• No issues found!
```

</details>

Maybe this will be fixed by [#54776](https://github.com/flutter/flutter/issues/54776)

Thank you

**Comment by Levi-Lesches** [comment]
> Maybe this will be fixed by #54776

From what I've seen, it looks like it will. 

Also, this is probably a duplicate of #19623, which closed in favor of #54776, so maybe the same should be done here too.

**Comment by TahaTesser** [comment]
Hi @lazylazyllama 
Closing in favor of [#54776](https://github.com/flutter/flutter/issues/54776)
Thank you for confirmation
